### PR TITLE
#3822 Add patients not always editable during vaccine dispensing

### LINKED ISSUE #3822: Patients should not default to being editable on vaccine dispensing
## Describe the bug

At the moment there is no check on home store or 'edit anywhere' pref for patients being edited through vaccine dispening

### To reproduce

1. Lookup a patient through the lookup api without having the edit anywhere pref on.
2. Click next and edit the patient

### Expected behaviour

Patients should only be editable if the mobile store is the patients home store or if the edit anywhere pref is on.

### Proposed Solution

At a check for a patient being editable

### Version and device info

- App version: 8.0 development
- Tablet model:
- OS version:

### Additional context

N/a

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3822 

## Change summary

- Made patients not editable unless the pref is set in vaccine dispensing
- was annoying to make it work, so I changed when selecting a patient from the lookup, to be saved straight away to the db. This changed the flow a fair amount, but I think simplifies it.

## Testing

- [ ] When the pref `can_edit_patients_from_any_store` is SET, you can edit patient details (DOES NOT MEAN EXTRA INFORMATION) for non-local patients during vaccine dispensing
- [ ] When the pref `can_edit_patients_from_any_store` is NOT SET, you can NOT edit patient details 
(DOES NOT MEAN EXTRA INFORMATION) for non-local patients during vaccine dispensing
- [ ] When the pref `can_edit_patients_from_any_store` is NOT SET, you can NOT edit patient details (DOES NOT MEAN EXTRA INFORMATION) for non-local patients during NORMAL dispensing
- [ ] When the pref `can_edit_patients_from_any_store` is SET, you can edit patient details (DOES NOT MEAN EXTRA INFORMATION) for non-local patients during NORMAL dispensing

### Related areas to think about

N/A

### CODE REVIEW

File: src/actions/Entities/NameNoteActions.js [hunk 1]
```diff
@@ -97,12 +99,9 @@ const updateForm = (data, validator) => ({
...
 
-const saveEditing = optionalNameID => (dispatch, getState) => {
-  const { nameID, patientEventID, ...nameNote } = selectCreatingNameNote(getState()) ?? {};
-
-  UIDatabase.write(() =>
-    createRecord(UIDatabase, 'NameNote', nameNote, patientEventID, nameID || optionalNameID)
-  );
+const saveEditing = () => (dispatch, getState) => {
+  const nameNote = selectCreatingNameNote(getState()) ?? {};
+  UIDatabase.write(() => createRecord(UIDatabase, 'NameNote', nameNote));
```
Comment (joshxg) [inline src/actions/Entities/NameNoteActions.js:104]: changed the `createRecord` to accept an object. The optionalNameID wasn't used

File: src/database/utilities/createRecord.js [hunk 2]
```diff
@@ -290,6 +306,9 @@ const createPatient = (database, patientDetails) => {
     nationality,
     ethnicity,
   });
+
+  nameNotes?.forEach(nameNote => createNameNote(database, nameNote));
```
Comment (joshxg) [inline src/database/utilities/createRecord.js:310]: running this when selecting a remote patient, so running this to create all the name notes pulled

File: src/actions/Entities/NameNoteActions.js [hunk 3]
```diff
@@ -55,36 +54,39 @@ const getMostRecentPCD = patient => {
...
-    // Only create name notes, never edit- ensure every new name note which is seeded has a new ID.
-    const id = generateUUID();
-    const asObject = { ...seedPCD.toObject(), id, entryDate: new Date().getTime() };
-    dispatch(select(asObject, isValid));
-  } else {
-    dispatch(select(seedPCD, isValid));
-  }
+  const isValid = validateData(jsonSchema, newNameNote.data);
+
+  dispatch(select(newNameNote, isValid));
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply src/actions/Entities/NameNoteActions.js:89]: yeah, I think so. that reads well to me.

**Comment by mark-prins** [comment]
side note: resulted in spotting this one https://github.com/sussol/msupply/issues/8379

**Review by mark-prins — APPROVED** [review]
LGTM!
That all works, and changes all look good 😃

**Pull request merged**